mm-bot/src/services/ethereum.py

```diff
@@ -117,40 +97,28 @@ def withdraw(deposit_id, dst_addr, amount, value):
     return tx_hash
 
 
-def create_withdraw(deposit_id, dst_addr_bytes, amount, value):
-    exceptions = []
-    for index, w3 in enumerate(w3_clients):
-        try:
-            unsent_tx = contracts_rpc[index].functions.withdraw(deposit_id, dst_addr_bytes, amount).build_transaction({
-                "chainId": ETH_CHAIN_ID,
-                "from": accounts_rpc[index].address,
-                "nonce": get_nonce(w3, accounts_rpc[index].address),
-                "value": value,
-            })
-            signed_tx = w3.eth.account.sign_transaction(unsent_tx, private_key=accounts_rpc[index].key)
-            return unsent_tx, signed_tx
-        except Exception as exception:
-            logger.warning(f"[-] Failed to create withdraw eth on node: {exception}")
-            exceptions.append(exception)
-    logger.error(f"[-] Failed to create withdraw eth on all nodes")
-    raise Exception(f"Failed to create withdraw eth on all nodes: [{', '.join(str(e) for e in exceptions)}]")
+@use_fallback(rpc_nodes, logger, "Failed to create withdraw eth")
+def create_withdraw(deposit_id, dst_addr_bytes, amount, value, rpc_node=main_rpc_node):
+    unsent_tx = rpc_node.contract.functions.withdraw(deposit_id, dst_addr_bytes, amount).build_transaction({
+        "chainId": ETH_CHAIN_ID,
+        "from": rpc_node.account.address,
+        "nonce": get_nonce(rpc_node.w3, rpc_node.account.address),
+        "value": value,
+    })
+    signed_tx = rpc_node.w3.eth.account.sign_transaction(unsent_tx, private_key=rpc_node.account.key)
+    return unsent_tx, signed_tx
 
 
 def get_nonce(w3: Web3, address):
     return w3.eth.get_transaction_count(address)
 
 
-def estimate_gas_fee(transaction):
-    for w3 in w3_clients:
-        try:
-            gas_limit = w3.eth.estimate_gas(transaction)
-            fee = w3.eth.gas_price
-            gas_fee = fee * gas_limit
-            return gas_fee
-        except Exception as exception:
-            logger.warning(f"[-] Failed to estimate fee on node: {exception}")
-    logger.error(f"[-] Failed to estimate fee on all nodes")
-    raise Exception("Failed to estimate fee on all nodes")
+@use_fallback(rpc_nodes, logger, "Failed to estimate gas fee")
+def estimate_gas_fee(transaction, rpc_node=main_rpc_node):
```

Comment:
What do you think of changing this to something more descriptive or adding a comment on what exactly this fee means?
I don't know if `estimate_gas_fee` is the correct term because this is just the overall fee of our operational cost. So `estimate_overall_fee` could be a better option.

Reply:
It is not the overall fee. It is only the fee for a single transaction. What do you think about call it `estimate_transaction_fee`?

Reply:
Sorry I got confused with the fee that takes into account everything. Yes `estimate_transaction_fee` looks like a good idea.

Reply:
Done